bridge/contracts/utils/ImmutableAuth.sol

```diff
@@ -531,3 +531,30 @@ abstract contract ImmutableETHDKG is ImmutableFactory {
         return 0x455448444b470000000000000000000000000000000000000000000000000000;
     }
 }
+
+abstract contract ImmutableBridgeRouter is ImmutableFactory {
+    address private immutable _bridgeRouter;
+    error OnlyBridgeRouter(address sender, address expected);
+
+    modifier onlyBridgeRouter() {
+        if (msg.sender != _bridgeRouter) {
+            revert OnlyBridgeRouter(msg.sender, _bridgeRouter);
+        }
+        _;
+    }
+
+    constructor() {
+        _bridgeRouter = getMetamorphicContractAddress(
+            0x427269646765526f757465720000000000000000000000000000000000000000,
```

Comment:
Magic address, should be brought out into a constant with descriptive comment around how to calculate it.